src/crab/array_domain.cpp

```
@@ -4,201 +4,73 @@
-    // Return true if [o, o + size) definitely overlaps with the cell,
-    // where o is a constant expression.
+    // Return true if [o, o + sz) definitely overlaps with the cell.
     [[nodiscard]]
-    bool overlap(const offset_t& o, const unsigned size) const {
-        const Interval x = to_interval();
-        const Interval y = to_interval(o, size);
-        const bool res = (!(x & y).is_bottom());
-        CRAB_LOG("array-expansion-overlap",
-                 std::cout << "**Checking if " << x << " overlaps with " << y << "=" << res << "\n";);
-        return res;
+    bool overlap(const offset_t o, const unsigned sz) const {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)